thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/MultiThreadView.kt

```diff
@@ -169,81 +175,66 @@ class MultiThreadView(
-            thread.state in stateList.checkBoxListSelectedValues
+    private fun filter(thread: Thread?): Boolean {
+        if (thread == null) {
+            return false
         }
-        add { thread ->
-            thread.system in systemList.checkBoxListSelectedValues
-        }
-        add { thread ->
-            thread.pool in poolList.checkBoxListSelectedValues
+
+        if (thread.state.name !in stateList.checkBoxListSelectedValues
+            || thread.system !in systemList.checkBoxListSelectedValues
+            || thread.pool !in poolList.checkBoxListSelectedValues) {
+            return false
         }
-        add { thread ->
-            val query = searchField.text ?: return@add true
 
-            val idMatches = thread.id.toString().contains(query)
-            val nameMatches = thread.name.contains(query, ignoreCase = true)
-            val systemMatches = thread.system != null && thread.system.contains(query, ignoreCase = true)
-            val scopeMatches = thread.scope != null && thread.scope.contains(query, ignoreCase = true)
-            val stateMatches = thread.state.name.contains(query, ignoreCase = true)
-            val stackMatches = thread.stacktrace.any { stack -> stack.contains(query, ignoreCase = true) }
+        val query = searchField.text ?: return true
 
-            idMatches || nameMatches || systemMatches || scopeMatches || stateMatches || stackMatches
-        }
+        return thread.id.toString().contains(query)
+            || thread.name.contains(query, ignoreCase = true)
+            || thread.system != null && thread.system.contains(query, ignoreCase = true)
+            || thread.scope != null && thread.scope.contains(query, ignoreCase = true)
+            || thread.state.name.contains(query, ignoreCase = true)
+            || thread.stacktrace.any { stack -> stack.contains(query, ignoreCase = true) }
     }
 
     private fun updateData() {
         BACKGROUND.launch {
-            val selectedThreadDumps = threadDumps.filterIndexed { index, _ ->
-                index + 1 in threadDumpCheckboxList.checkBoxListSelectedIndices
+            val filteredThreadDumps = currentLifespanList.filter { lifespan ->
+                lifespan.any(::filter)
             }
 
-            val filteredThreadDumps = selectedThreadDumps.map {
-                ThreadDump(
-                    it.version,
-                    it.threads.filter { thread ->
-                        filters.all { filter -> filter(thread) }
-                    }
-                )
-            }
-            val data = filteredThreadDumps.toLifespanList()
-
             EDT_SCOPE.launch {
-                if (mainTable.selectionModel.isSelectionEmpty) {
-                    val newModel = ThreadModel(data)
-                    mainTable.installColumnFactory(newModel.columns, false)
-                    mainTable.model = newModel
-                    mainTable.createDefaultColumnsFromModel()
-                } else {
```

Comment:
Now there's only one place a new `ThreadModel` gets created, instead of duplicate code in two branches